op-defender/psp_executor/defender.go

```diff
@@ -0,0 +1,290 @@
+func FetchAndExecute() {
+	//1. Fetch the privatekey of the account in GCP secret Manager.
+	privatekey, err := FetchPrivateKeyInGcp()
+	if err != nil {
+		log.Crit("Failed to fetch the privatekey from GCP secret manager: %v", err.Error())
+	}
+	// 2. Fetch the CORRECT Nounce PSP in the GCP secret Manager and return the data to execute.
+	superchainconfig_address, safe_address, data, err := FetchPSPInGCP()
+	if err != nil {
+		log.Crit("Failed to fetch the PSP from GCP secret manager: %v", err.Error())
+	}
+
+	// 3. Get the L1 client and ensure RPC is correct.
+	l1client, err := GetTheL1Client()
+	if err != nil {
+		log.Crit("Failed to get the L1client: %v", err.Error())
+	}
+
+	// 4. Execute the PSP on the chain.
+	ctx := context.Background() //TODO: Check if we really do need to context if yes we will keep it otherwise we will remove this.
+	PspExecutionOnChain(ctx, l1client, superchainconfig_address, privatekey, safe_address, data)
+}
+
+// GetTheL1Client() will return the L1 client based on the RPC provided in the config and ensure that the RPC is not production one.
+func GetTheL1Client() (*ethclient.Client, error) {
+	client, err := ethclient.Dial(LocalhostRPC) //Need to change this to the correct RPC (mainnet or sepolia) but for now hardcoded to localhost.
+	if LocalhostRPC != "http://localhost:8545" {
+		log.Warn("This is not the RPC localhost are you sure you want to continue (yes/no)")
+		var response string
+		fmt.Scanln(&response)
+		if response != "yes" {
+			log.Crit("Not yes, We Exiting the program.")
+		}
+	}
+	if err != nil {
+		log.Crit("Failed to connect to the Ethereum client: %v", err.Error())
+	}
+	return client, nil
+}
+
+// FetchPrivateKey() will fetch the privatekey of the account that will execute the pause (from the GCP secret manager).
+func FetchPrivateKeyInGcp() (string, error) {
+	return "2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6", nil // Mock with a well-known private key from test test ... test junk derivation (9).
+}
+
+// FetchPSPInGCP() will fetch the correct PSPs into GCP and return the Data.
+func FetchPSPInGCP() (string, string, []byte, error) { //superchainconfig_address, safe_address, data, error
+	// need to fetch check first the nonce with the same method with `checkPauseStatus` and then return the data for this PSP.
+
+	return "0xC2Be75506d5724086DEB7245bd260Cc9753911Be", "0x4141414142424242414141414242424241414141", []byte{0x41, 0x42, 0x43}, nil //errors.New("Not implemented") mock with simple value to make a call on L1.
+}
+
+// PSPexecution(): PSPExecutionOnChain is a core function that will check that status of the superchain is not paused and then send onchain transaction to pause the superchain.
+func PspExecutionOnChain(ctx context.Context, l1client *ethclient.Client, superchainconfig_address string, privatekey string, safe_address string, data []byte) {
+	fmt.Println("PSP Execution Pause")
```

Comment:
Should use log functionality instead of println around

Reply:
That's correct thanks good catch!